lib/tasks/republish.rake

```diff
@@ -0,0 +1,8 @@
+desc "Republish all documents with non-english html attachments"
+task :republish_non_english_html_attachments do
+  Edition.where(state: %[published draft]).where.not(attachables: []).each do |edition|
```

Comment:
Trying this out in the app console, looks like you need a `%w` instead of a `%`.

```suggestion
  Edition.where(state: %w[published draft]).where.not(attachables: []).each do |edition|
```

Also, `Edition.where(attachables: [])` returns an empty array, and `Edition.where(attachables: nil)` gives `ActiveRecord::StatementInvalid (Mysql2::Error: Unknown column 'editions.attachables' in 'where clause')`. I suspect some sort of [joins](https://guides.rubyonrails.org/active_record_querying.html#joins) method will be needed before we can query this,  but because of the [weird way we save `attachments` in Whitehall](https://github.com/alphagov/whitehall/blob/7f0b913e2f3753864701d71f2c74c8f59bf225cb/app/models/attachable.rb#L18-L26), I'm not quite sure how to go about that.

Reply:
The other option would be to select everything and skip them in the ruby code — presumably a bit slower but I wonder if it would be significant, when it's a one-off task (and compared to the time needed for the republications themselves).

Reply:
I think I can do something like this:

```ruby
HtmlAttachment.where(attachable_type: 'Edition').where.not(deleted: true).joins("JOIN editions ON editions.id = attachments.attachable_id").where.not(editions: {primary_locale: 'en'})
```

But that returns only six results, which seems doubtful.